src/persist/src/client.rs

```diff
@@ -78,175 +81,197 @@ impl RuntimeClient {
         name: &str,
     ) -> (StreamWriteHandle<K, V>, StreamReadHandle<K, V>) {
         let (tx, rx) = PFuture::new();
-        self.core.send(RuntimeCmd::IndexedCmd(Cmd::Register(
-            name.to_owned(),
-            (K::codec_name(), V::codec_name()),
-            tx,
-        )));
-        let id = rx.recv();
-        let write = StreamWriteHandle::new(name.to_owned(), id.clone(), self.clone());
-        let meta = StreamReadHandle::new(name.to_owned(), id, self.clone());
-        (write, meta)
+        self.sender
+            .send_runtime_cmd(RuntimeCmd::IndexedCmd(Cmd::Register(
+                name.to_owned(),
+                (K::codec_name(), V::codec_name()),
+                tx,
+            )));
+        let stream_id = rx.recv();
+
+        let write = StreamWriteHandle {
+            name: name.to_owned(),
+            stream_id: stream_id.clone(),
+            client: self.clone(),
+            _phantom: PhantomData,
+        };
+        let read = StreamReadHandle {
+            name: name.to_owned(),
+            stream_id,
+            client: RuntimeReadClient {
+                sender: CmdReadSender::Full(self.sender.clone()),
+                stopper: self.stopper.clone(),
```

Comment:
nit: I prefer `Arc::clone()`, because it makes it more obvious that this is not a "real" clone. But: we could enable a lint for this and we don't, so ... :man_shrugging: